code/game/objects/structures/tables_racks.dm

```diff
@@ -545,6 +554,72 @@
 /obj/structure/table/reinforced/brass/ratvar_act()
 	obj_integrity = max_integrity
 
+/obj/structure/table/tray
+	name = "surgical tray"
+	desc = "A small metal tray with wheels."
+	anchored = FALSE
+	smooth = SMOOTH_FALSE
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "tray"
+	buildstack = /obj/item/stack/sheet/mineral/titanium
+	buildstackamount = 2
+	var/list/typecache_can_hold = list(/mob, /obj/item)
+	var/list/held_items = list()
+
+/obj/structure/table/tray/Initialize()
+	. = ..()
+	verbs -= /obj/structure/table/verb/do_flip
+	typecache_can_hold = typecacheof(typecache_can_hold)
+	for(var/atom/movable/held in get_turf(src))
+		if(is_type_in_typecache(held, typecache_can_hold))
+			held_items += held.UID()
+
+/obj/structure/table/tray/Move(NewLoc, direct)
+	var/atom/OldLoc = loc
+
+	. = ..()
+	if(!.) // ..() will return 0 if we didn't actually move anywhere.
+		return .
+
+	if(direct & (direct - 1)) // This represents a diagonal movement, which is split into multiple cardinal movements. We'll handle moving the items on the cardinals only.
+		return .
+
+	playsound(loc, pick('sound/items/cartwheel1.ogg', 'sound/items/cartwheel2.ogg'), 100, 1, ignore_walls = FALSE)

```

Comment:
Might as well add this to things like rollerbeds, janicarts and such while you’re at it.